src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexManager.cs

```diff
@@ -249,23 +257,42 @@ private BaseDirectory CreateDirectory(string indexName)
 
         private void Write(string indexName, Action<IndexWriter> action, bool close = false)
         {
-            var writer = _writers.GetOrAdd(indexName, name =>
+            if (!_writers.TryGetValue(indexName, out var writer))
             {
-                var directory = CreateDirectory(indexName);
-                var config = new IndexWriterConfig(LuceneVersion, new StandardAnalyzer(LuceneVersion))
+                lock (this)
```

Comment:
Good catch preventing a second writer getting in, while keeping an optimized code path for the common case.

Reply:
Thanks, exactly, just re-tried seems to work well.